src/SeaIceMomentumEquations/sea_ice_external_stress.jl

```diff
@@ -0,0 +1,105 @@
+using Adapt
+using Oceananigans.Fields: ZeroField
+
+# Fallback
+@inline implicit_τx_coefficient(i, j, k, grid, stress, clock, fields) = zero(grid)
+@inline implicit_τy_coefficient(i, j, k, grid, stress, clock, fields) = zero(grid)
+
+# Fallback
+@inline explicit_τx(i, j, k, grid, stress, clock, fields) = zero(grid)
+@inline explicit_τy(i, j, k, grid, stress, clock, fields) = zero(grid)
+
+@inline explicit_τx(i, j, k, grid, stress::Number, clock, fields) = stress
+@inline explicit_τy(i, j, k, grid, stress::Number, clock, fields) = stress
+
+@inline explicit_τx(i, j, k, grid, stress::AbstractArray, clock, fields) =  @inbounds stress[i, j, k] 
+@inline explicit_τy(i, j, k, grid, stress::AbstractArray, clock, fields) =  @inbounds stress[i, j, k] 
+
+# NamedTuple stess (assuming it is `u` and `v`)
+@inline implicit_τx_coefficient(i, j, k, grid, stress::NamedTuple, clock, fields) = implicit_τx_coefficient(i, j, k, grid, stress.u, clock, fields)
+@inline implicit_τy_coefficient(i, j, k, grid, stress::NamedTuple, clock, fields) = implicit_τy_coefficient(i, j, k, grid, stress.v, clock, fields)
+
+@inline explicit_τx(i, j, k, grid, stress::NamedTuple, clock, fields) = explicit_τx(i, j, k, grid, stress.u, clock, fields)
+@inline explicit_τy(i, j, k, grid, stress::NamedTuple, clock, fields) = explicit_τx(i, j, k, grid, stress.v, clock, fields)
+
+#####
+##### SemiImplicitStress
+#####
+
+struct SemiImplicitStress{U, V, FT}
```

Comment:
I don't completely understand the necessity of this. I understand we have to treat the stress implicitly. But why can't we simply implement this numerically if given explicit stress --- ie compute a drag coefficient, treat the term implicitly, and thne add the residual at the end?

Reply:
Right, this was to allow also explicit stresses (I guess we never want to do that). With this we can also do semi-implicit sea-ice atmosphere stresses, which maybe also is not super important. I can open an issue where we can decide how we treat external stresses, if we want to always have a semi-implicit stress or allow other types of stress formulation